.github/workflows/ci_linux_x64_clang_tsan.yml

```diff
@@ -54,4 +54,11 @@ jobs:
           # Use a modern clang explicitly.
           CC: clang-19
           CXX: clang++-19
-        run: ./build_tools/cmake/build_and_test_tsan.sh
+          SCCACHE_AZURE_CONNECTION_STRING: "${{ secrets.AZURE_CCACHE_CONNECTION_STRING }}"
+          SCCACHE_AZURE_BLOB_CONTAINER: ccache-container
+          SCCACHE_CACHE_ZSTD_LEVEL: 10
+          SCCACHE_AZURE_KEY_PREFIX: "ci_linux_x64_clang_tsan"
+        run: |
+          source build_tools/cmake/setup_sccache.sh
+          ./build_tools/cmake/build_and_test_tsan.sh
+          sccache --show-stats
```

Comment:
Low cache hit rates here too: https://github.com/iree-org/iree/actions/runs/10836779139/job/30071285271?pr=18489#step:5:14493
```
Compile requests                    5369
Compile requests executed           5369
Cache hits                           814
Cache hits (C/C++)                   814
Cache misses                        4540
Cache misses (C/C++)                4540
```

Nightly builds with GCP and ccache can be similarly low for TSan after code changes: https://github.com/iree-org/iree/actions/runs/10828056581/job/30042565704#step:3:9616
```
Cacheable calls:   5351 / 5369 (99.66%)
  Hits:             798 / 5351 (14.91%)
```

but ideally they are high: https://github.com/iree-org/iree/actions/runs/10808905140/job/29982886219#step:3:9589
```
Cacheable calls:   5334 / 5350 (99.70%)
  Hits:            5221 / 5334 (97.88%)
```

Reply:
Better on a previous run: https://github.com/iree-org/iree/actions/runs/10836229607/job/30069537801#step:5:14457

Reply:
> Low cache hit rates here too: https://github.com/iree-org/iree/actions/runs/10836779139/job/30071285271?pr=18489#step:5:14493
> 
> ```
> Compile requests                    5369
> Compile requests executed           5369
> Cache hits                           814
> Cache hits (C/C++)                   814
> Cache misses                        4540
> Cache misses (C/C++)                4540
> ```
> 
> Nightly builds with GCP and ccache can be similarly low for TSan after code changes: https://github.com/iree-org/iree/actions/runs/10828056581/job/30042565704#step:3:9616
> 
> ```
> Cacheable calls:   5351 / 5369 (99.66%)
>   Hits:             798 / 5351 (14.91%)
> ```
> 
> but ideally they are high: https://github.com/iree-org/iree/actions/runs/10808905140/job/29982886219#step:3:9589
> 
> ```
> Cacheable calls:   5334 / 5350 (99.70%)
>   Hits:            5221 / 5334 (97.88%)
> ```

this was my most recent run https://github.com/iree-org/iree/pull/18489#discussion_r1757553211